### ISSUE TITLE
Not able to run simple.cfg

### ISSUE THREAD

**Issue by abhishekk06** [wangziqi2013/zsim-base#2]
Hi @wangziqi2013,

I am able to compile code but for some reason not able to run simple.cfg (I was hitting same issue in original zsim). Given below is the console print from zsim. Can you please help?

(zsim) zsim-base$ ./build/opt/zsim tests/simple.cfg
[H] Starting zsim, built Wed Mar 24 23:57:22 EDT 2021 (rev )
[H] Creating global segment, 1024 MBs
[H] Global segment shmid = 229388
[H] Deadlock detection ON
A: Source/pin/injector_nonmac/auxvector.cpp: CopyAux: 291: unexpected AUX VEC type 26

################################################################################
################################################################################
addr2line -C -f -e "/home/mdl/azk6085/CSE530/zsim-base/pin-2.14/intel64/bin/pinbin" 0x5555557147f9 0x5555557155ce 0x5555557158a0 0x555555937b3c 0x55555593d2ff 0x55555593ee2d 0x55555592efe1 0x555555933b2d 0x55555592422e 0x55555592e230 0x7ffff700b555 0x55555570917a
A: Source/pin/injector_nonmac/auxvector.cpp: CopyAux: 291: unexpected AUX VEC type 26

################################################################################
################################################################################
addr2line -C -f -e "/home/mdl/azk6085/CSE530/zsim-base/pin-2.14/intel64/bin/pinbin" 0x5555557147f9 0x5555557155ce 0x5555557158a0 0x555555937b3c 0x55555593d2ff 0x55555593ee2d 0x55555592efe1 0x555555933b2d 0x55555592422e 0x55555592e230 0x7ffff700b555 0x55555570917a
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
Detach Service Count: 1
Pin 2.14
Copyright (c) 2003-2015, Intel Corporation. All rights reserved.
@CHARM-VERSION: $Rev: 71293 $
@CHARM-BUILDER: BUILDER
@CHARM-COMPILER: gcc 4.4.7
@CHARM-TARGET: ia32e
@CHARM-CFLAGS:  __OPTIMIZE__=1  __NO_INLINE__=__NO_INLINE__
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
??
??:0
Detach Service Count: 1
Pin 2.14
Copyright (c) 2003-2015, Intel Corporation. All rights reserved.
@CHARM-VERSION: $Rev: 71293 $
@CHARM-BUILDER: BUILDER
@CHARM-COMPILER: gcc 4.4.7
@CHARM-TARGET: ia32e
@CHARM-CFLAGS:  __OPTIMIZE__=1  __NO_INLINE__=__NO_INLINE__
C: Injector exited with signal 6
E: Wait for injector failed: No child processes
[H] Child 5768 done
C: Injector exited with signal 6
E: Wait for injector failed: No child processes
[H] Child 5767 done
[H] All children done, exiting

**Comment by wangziqi2013** [comment]
I think this is an OS version issue, basically, PIN failed a kernel version check.

The solution is to add "-ifeellucky" in the sim options: Open simple.cfg, find sim.pinOptions, add "-ifeellucky" before or after the injection command, and try again.

It this proves to be working, you may submit a pull request to merge the change (or I can do it instead)

**Comment by wangziqi2013** [comment]
I just updated simple.cfg, you can check it out and see if it works.

**Comment by abhishekk06** [comment]
Hi @wangziqi2013 ,

Thanks for the quick help. I think I have get past that issue but now I am seeing ".so" not found. Any suggestion. I do have the libhdf5 installed in my env. 

![image](https://user-images.githubusercontent.com/55446334/112418741-7527ef80-8d00-11eb-8691-6c6bd3a32941.png)

**Comment by abhishekk06** [comment]
Any idea how can I verify that?

**Comment by wangziqi2013** [comment]
Ok, this seems unusual. Maybe you need to verify that by running "find / -name "libhdf5.so.10"" and see if any result shows up. Normally, if zsim links successfully then there shouldn't be such an issue.

If the find commands actually finds the library, and it is in a non-standard location, you could manually force the run-time dynamic linker to search the location with LD_LIBRARY_PATH. Just add "LD_LIBRARY_PATH=[your location]" before the zsim command.

**Comment by abhishekk06** [comment]
Ok. I was not able to use "Find "  or "dpkg" to check if I have the hdf5 installed or not but I went ahead and manually set LD_LIBRARY_PATH (till $HOME/hdf5-1.8.22/lib/) and now I can get the first test running) :) 

![image](https://user-images.githubusercontent.com/55446334/112421396-511add00-8d05-11eb-81d1-3a4f2aa1f4f3.png)


Thank you very much for your quick help and also for taking the initiative of setting up a much more stable zSim repo. :)
 I also went through your [GHC 6023 blogs](https://wangziqi2013.github.io/article/2020/01/01/zsim-microarch.html) and I feel might need your expert advice to learn more about zSim and get fully comfortable. It will be great if we can connect on[ LinkedIn ](https://www.linkedin.com/in/abhishekk105/) or [email](abhishekk105@gmail.com) so that I can ping you in case of a blocker issue.
